src/ompl/tools/thunder/SPARSdb.h

```diff
@@ -740,6 +756,11 @@ namespace ompl
                 return si_->distance(stateProperty_[a], stateProperty_[b]);
             }
 
+            double costFunction(const Vertex a, const Vertex b) const
+            {
+                return pdef_->getOptimizationObjective()->motionCost(stateProperty_[a], stateProperty_[b]).value();
+            }
```

Comment:
Consider passing this in.  Also naming, costFunction should be motionCostFuntion I think as edges length is a cost, just a different one; no?

Reply:
I see what you mean, I can name it costObjectiveFunction since it will be using the custom cost objective vs path length.
Motion cost is what ompl uses to denote the cost of a segment, it does not indicate which cost function you are using.

Reply:
What do you mean that I should pass the function in? Make an enum of functions?